src/EditorFeatures/Test/Diagnostics/MockDiagnosticService.cs

```diff
@@ -32,15 +33,35 @@ public MockDiagnosticService()
         {
         }
 
-        public ImmutableArray<DiagnosticData> GetDiagnostics(Workspace workspace, ProjectId? projectId, DocumentId? documentId, object? id, bool includeSuppressedDiagnostics, bool forPullDiagnostics, CancellationToken cancellationToken)
+        public ImmutableArray<DiagnosticData> GetPullDiagnostics(Workspace workspace, ProjectId? projectId, DocumentId? documentId, object? id, bool includeSuppressedDiagnostics, Option2<DiagnosticMode> diagnosticMode, CancellationToken cancellationToken)
+        {
+            return GetDiagnostics(workspace, projectId, documentId);
+        }
+
+        public ImmutableArray<DiagnosticData> GetPushDiagnostics(Workspace workspace, ProjectId? projectId, DocumentId? documentId, object? id, bool includeSuppressedDiagnostics, Option2<DiagnosticMode> diagnosticMode, CancellationToken cancellationToken)
```

Comment:
clearly named overloads have been added so that clients can say if they want diagnostics for push or pull purposes.